src/Evolution/DiscontinuousGalerkin/SlopeLimiters/HwenoModifiedSolution.hpp

```diff
@@ -0,0 +1,497 @@
+ * 2. The distance between the cell average of the modified solution and the
+ *    cell average of the original solution on each secondary neighbor. Note
+ *    however that one secondary neighbor (or, rarely, several) is be excluded
+ *    from this minimization: the secondary neighbor(s) where the original
+ *    solution has the most different cell average from the local element. This
+ *    helps to prevent an outlier (e.g., near a shock) from biasing the fit.
+ *
+ * Simultaneously, the modified solution must satisfy the constraint that
+ * 3. The cell average of the modified solution on the local element must equal
+ *    the cell average of the local element's original solution.
+ *
+ * Below we give the mathematical form of the constraints described above and
+ * show how these are translated into a numerical algorithm.
+ *
+ * Consider an element \f$I_0\f$ with neighbors \f$I_1, I_2, ...\f$. For a
+ * given tensor component \f$u\f$, the values on each of these elements are
+ * \f$u_0, u_1, u_2, ...\f$. Taking for the sake of example the primary
+ * neighbor to be \f$I_1\f$, the modified solution \f$\phi\f$ must minimize
+ *
+ * \f[
+ * \chi^2 = \int_{I_1} (\phi - u_1)^2 dV
+ *          + \sum_{l} \left( \int_{I_l} ( \phi - u_l ) dV \right)^2,
+ * \f]
+ *
+ * subject to the constaint
+ *
+ * \f[
+ * C = \int_{I_0} ( \phi - u_0 ) dV = 0.
+ * \f]
+ *
+ * where \f$l\f$ ranges over the secondary neighbors, but excludes the one
+ * (or more) secondary neighbor(s) where the mean of \f$u\f$ is the most
+ * different from the mean of \f$u_0\f$. Note that in 1D, this implies that
+ * \f$l\f$ ranges over the empty set; for each modified solution, one neighbor
+ * is the primary neighbor and the other is the excluded neighbor.
+ *
+ * The integrals are evaluated by quadrature. We denote the quadrature weights
+ * by \f$\omega_s\f$ and the values of some data \f$X\f$ at the quadrature
+ * nodes by \f$X_s\f$. We use subscripts \f$r,s\f$ to denote quadrature nodes
+ * on the neighbor and local elements, respectively. The minimization becomes
+ *
+ * \f[
+ * \chi^2 = \sum_{r} \omega_{1,r} ( \phi_{r} - u_{1,r} )^2
+ *          + \sum_{l} \left( \sum_{r} \omega_{l,r} ( \phi_{r} - u_{l,r} )
+ *                     \right)^2,
+ * \f]
+ *
+ * subject to the constraint
+ *
+ * \f[
+ * C = \sum_{s} \omega_{0,s} ( \phi_{s} - u_{0,s} ) dV = 0.
+ * \f]
+ *
+ * Note that \f$\phi\f$ is a function defined on the local element \f$I_0\f$,
+ * and so is fully represented by its values \f$\phi_{s}\f$ at the quadrature
+ * points on this element. When evaluating \f$\phi\f$ on element \f$I_l\f$, we
+ * obtain the function values \f$\phi_{r}\f$ by polynomial extrapolation,
+ * \f$\phi_{r} = \sum_{s} M^{(l)}_{rs} \phi_{s}\f$, where \f$M^{(l)}_{rs}\f$ is
+ * the interpolation/extrapolation matrix that interpolates data defined at grid
+ * points \f$x_s\f$ and evaluates it at grid points \f$x_r\f$. Thus,
+ *
+ * \f[
+ * \chi^2 = \sum_{r} \omega_{1,r}
```

Comment:
Seeing this equation I think using the superscript for the element on all quantities would make the equations much easier to read :)